cmd/describe/breakglasscredential/cmd.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package breakglasscredential
+
+import (
+	"fmt"
+	"os"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/breakglasscredential"
+	"github.com/openshift/rosa/pkg/externalauthprovider"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/output"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "break-glass-credential",
+	Aliases: []string{"break-glass-credentials", "breakglasscredential", "breakglasscredentials"},
+	Short:   "Show details of a break glass credential on a cluster",
+	Long:    "Show details of a break glass credential on a cluster.",
+	Example: `  # Show details of a break glass credential with ID "12345" on a cluster named "mycluster"
+  rosa describe break-glass-credential 12345 --cluster=mycluster `,
+	Run:    run,
+	Hidden: true,
+	Args:   cobra.MaximumNArgs(1),
+}
+
+var args struct {
+	id string
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.SortFlags = false
+	ocm.AddClusterFlag(Cmd)
+	output.AddFlag(Cmd)
+	flags.StringVar(
+		&args.id,
+		"id",
+		"",
+		"Id for the break glass credential of the cluster to target",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+	err := runWithRuntime(r, cmd, argv)
+	if err != nil {
+		r.Reporter.Errorf(err.Error())
+		os.Exit(1)
+	}
+}
+
+func runWithRuntime(r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+	breakGlassCredentialId := args.id
+	// Allow the use also directly set the break glass credential id as positional parameter
+	if len(argv) == 1 && !cmd.Flag("id").Changed {
+		breakGlassCredentialId = argv[0]
+	}
+	if breakGlassCredentialId == "" {
+		return fmt.Errorf("you need to specify a break glass credential id with '--id' parameter")
+	}
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+
+	externalAuthService := externalauthprovider.NewExternalAuthService(r.OCMClient)
+	err := externalAuthService.IsExternalAuthProviderSupported(cluster, clusterKey)
+	if err != nil {
+		return err
+	}
+
+	r.Reporter.Debugf("Fetching the break glass credential '%s' for cluster '%s'", breakGlassCredentialId, clusterKey)
+	r.Reporter.Infof(
```

Comment:
@chenz4027 What's the use-case here? We're making an assumption that people have `jq` available to them. I personally don't think we should rely on additional tools for people to use `rosa` for key workflows. For example , does `jq` exist for Windows?

Reply:
chiming it as I suggested it, the use case is that this command provides a kubeconfig for the user.
Showing it in the describe is not great UX IMO.
Here we are providing it to the user in the json output and an hint on how to retrieve it with jq (but no strict dependency on it).

I was also looking at AKS, https://learn.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-get-credentials, they provide a much more opinionated default (writing kubeconfig to your default .kube/config) if we want to follow that.

Reply:
@chenz4027 @andreadecorte Is this exposed to customers or is Red Hat SRE the end-user?

Reply:
I think it's for the end-user, but will wait for @andreadecorte to confirm

Reply:
end user.
User flow is:
1. Create cluster with external authentication enabled
2. Use this command to get a kubeconfig
3. Use the kubeconfig to apply its own config for access
4. Profit

Reply:
@chenz4027 @andreadecorte: @gdbranco @ciaranRoche any myself have had a chat about this. We're against printing out that the user should use `jq` to get details of the `kubeconfig` to login. Reading through the epic, it feels like this is a core workflow, so should be as easy as possible for customers to achieve using only `rosa`.

We've come up with this proposal:

- `rosa create break-glass-credential -c rblake-test -id bar`
  - This would poll until the `kubeconfig` is created and print it as STDOUT at the end of the command
  - Ideally we're targeting the workflow for a customer to be able to do `rosa create break-glass-credential -c rblake-test -id bar > /path/to/kubeconfig`
- `rosa describe break-glass-credential`
   - Supports a `--kubeconfig` flag that replicates the output from the `rosa create break-glass-credential` command
   - Allows a user to do `rosa describe break-glass-credential --kubeconfg > /path/to/kubeconfig` to easily grab the credentials if they forgot them from the `create command`
 - `rosa list break-glass-credential`
 - `rosa delete break-glass-credential`

WDYT?

Reply:
Thanks @robpblake , an additional --kubeconfig would work and we could remind the user of that parameter when they do `rosa describe breakglasscredential` like what we have here. @andreadecorte WDYT?

Reply:
sounds like a solid proposal.
+1 on Maggie's idea to keep an hint for the user in the standard describe to use the --kubeconfig flag if he wants to get the kubeconfig, I think it helps on the discoverability

Reply:
Thanks @chenz4027 @andreadecorte 👍 

I've updated the proposal to capture @chenz4027 suggestion that the `rosa describe break-glass-credential` command includes the suggestion of the `--kubeconfig` flag to just to get the KubeConfig. @chenz4027 Are you happy to re-work this PR to meet that proposal? If yes, I'd suggest documenting it in the JIRA as well so that QE are clear on what needs to be tested.

cheers,

Rob

Reply:
Thanks @robpblake , yep I'll be working on it. Let me update the Jira and tag you ;)

Reply:
@robpblake Added one more parameter for kubeconfig. Would you mind taking another look?